HopsanGUI/Widgets/TextEditorWidget.cpp

```diff
@@ -589,7 +609,59 @@ void TextEditor::wheelEvent(QWheelEvent* event)
 //! @brief Updates list of auto complete words
 void TextEditor::updateAutoCompleteList()
 {
-    mpCompleter->setModel(new QStringListModel(gpTerminalWidget->mpHandler->getAutoCompleteWords(), mpCompleter));
+    if(mLanguage == HighlighterTypeEnum::Hcom) {
+        mpCompleter->setModel(new QStringListModel(gpTerminalWidget->mpHandler->getAutoCompleteWords(), mpCompleter));
+    }
+    else if(mLanguage == HighlighterTypeEnum::Cpp) {
+        QStringList lines = toPlainText().split("\n");
+
+        QStringList dataTypes = QStringList() << "size_t" << "double" << "int" << "SecondOrderTransferFunction" << "FirstOrderTransferFunction" << "Port";
+        QStringList functions = QStringList() << "addInputVariable" << "addOutputVariable" << "addConstant" << "addPowerPort" << "getSafeNodeDataPtr";
+
+        int bracketCounter=-1;
+
+        QStringList variables;
+        foreach(const QString &line, lines)
+        {
+            if(line.simplified().startsWith("class "))
+                bracketCounter = 0;
+
+            bracketCounter += line.count("{");
+            bracketCounter -= line.count("}");
+            if(bracketCounter != 1)
+                continue;
+
+            if(line.contains("()")) //Ignore functions
+                continue;
+
+            if(dataTypes.contains(line.simplified().section(" ",0,0)))
+            {
+                variables.append(line.simplified().split(","));
+            }
```

Comment:
What will happen if you initialize like this;
double apa = ko + 34

Will 34 become a "auto-complete variable"
Not such a big deal I guess

Reply:
I added a check that removes assignments, i.e. only keeps the string before the equality sign. That will for example convert "int apa = 42, kanin = ko + 34" to just "apa" and "kanin".